### PR TITLE
chore(ci): check for associated issue on pull request

### PR DESCRIPTION
**Pull request by strantalis**
resolves https://github.com/opentdf/platform/issues/1252

### CODE REVIEW

**Comment by strantalis** [comment]
test comment

**Comment by strantalis** [comment]
Example of the workflow working. 

https://github.com/strantalis/platform/pull/1

File: .github/workflows/checks.yaml [hunk 1]
```diff
@@ -56,6 +56,96 @@ jobs:
             authz
             examples
 
+  associated-issue:
+    name: Issue Accociated with Pull Request
+    if: contains(fromJSON('["pull_request", "pull_request_target"]'), github.event_name)
```
Comment (dmihalcik-virtru) [inline .github/workflows/checks.yaml:61]:
```suggestion
    if: >-
      contains(fromJSON('["pull_request", "pull_request_target"]'), github.event_name)
      && !contains(fromJSON('["dependabot", "opentdf-automation"]'), github.actor)
```

**Review by dmihalcik-virtru — CHANGES_REQUESTED** [review]
I haven't tested my change so please double check it

File: .github/workflows/checks.yaml (hunk 1 above)
Comment (strantalis) [reply .github/workflows/checks.yaml]: I updated the if condition with what I think will work. Not an easy way to test this until we merge in I think.

Comment (strantalis) [reply .github/workflows/checks.yaml]: I updated the if condition with what I think will work. Not an easy way to test this until we merge in I think.

**Review by dmihalcik-virtru — APPROVED** [review]

**Pull request merged**